pkg/types/currency_pair.go

```diff
@@ -18,9 +20,123 @@ func NewCurrencyPair(base, quote string) CurrencyPair {
+	}
+
+	if len(cp.Quote) > MaxCPFieldLength {
+		return fmt.Errorf("quote currency pair has wrong length")
+	}
+
+	return nil
+}
+
 // ValidateBasic checks that the Base / Quote strings in the CurrencyPair are formatted correctly, i.e.
-// Base + Quote are non-empty, and are in upper-case.
+// For base and quote asset:
+// check if the asset is formatted in the legacy validation form:
+//   - if so, check that fields are not empty and all upper case
+//   - else, check that the format is in the following form: tokenName,tokenAddress,chainID
 func (cp *CurrencyPair) ValidateBasic() error {
+	if err := cp.coreValidate(); err != nil {
+		return err
+	}
+
+	if IsLegacyAssetString(cp.Base) {
+		err := ValidateLegacyAssetString(cp.Base)
+		if err != nil {
+			return fmt.Errorf("base asset '%s' is invalid: %w", cp.Base, err)
+		}
+	} else {
+		err := ValidateDefiAssetString(cp.Base)
+		if err != nil {
+			return fmt.Errorf("base defi asset '%s' is invalid: %w", cp.Base, err)
+		}
+	}
+
+	// check quote asset
+	if IsLegacyAssetString(cp.Quote) {
+		err := ValidateLegacyAssetString(cp.Quote)
+		if err != nil {
+			return fmt.Errorf("quote asset '%s' is invalid: %w", cp.Quote, err)
+		}
+	} else {
+		err := ValidateDefiAssetString(cp.Quote)
+		if err != nil {
+			return fmt.Errorf("quote defi asset '%s' is invalid: %w", cp.Quote, err)
+		}
+	}
+
+	return nil
+}
+
+// ValidateLegacyAssetString checks if all fields in the asset string are formatted correctly, i.e.
+// - fields are fully uppercase
+// - fields
+//
+// NOTE: this function assumes that coreValidate() has already been run.
+func ValidateLegacyAssetString(asset string) error {
+	// check formatting of asset
+	if strings.ToUpper(asset) != asset {
+		return fmt.Errorf("incorrectly formatted asset string, expected: %s got: %s", strings.ToUpper(asset), asset)
+	}
+
+	split := strings.Split(asset, fieldSeparator)
+	if len(split) != 1 {
+		return fmt.Errorf("incorrectly formatted asset string, asset %s should not contain the %q character", asset, fieldSeparator)
+	}
+
+	return nil
+}
+
+// ValidateDefiAssetString checks that the asset string is formatted properly as a defi asset (tokenName,tokenAddress,chainID)
+// - check that the length of fields separated by fieldSeparator is expectedSplitLength
+// - check that the first split (tokenName) is formatted properly as a LegacyAssetString.
+//
+// NOTE: this function assumes that coreValidate() has already been run.
+func ValidateDefiAssetString(asset string) error {
+	split, err := splitDefiAssetString(asset)
+	if err != nil {
+		return err
+	}
+
+	// first element is a ticker, so we require it to pass legacy asset validation:
+	if err := ValidateLegacyAssetString(split[0]); err != nil {
+		return fmt.Errorf("asset field '%s' is invalid: %w", split[0], err)
+	}
+
+	return nil
+}
+
+// splitDefiAssetString splits a defi asset by the fieldSeparator and checks that it is the proper length.
+func splitDefiAssetString(asset string) ([]string, error) {
```

Comment:
nit: A public utility function that splits the string and returns some named string return values would be useful.